includes/functions.php

```diff
@@ -2064,13 +2080,17 @@ function device_is_up($device, $record_perf = false)
                     [$device['device_id'], $going_down]
                 );
             }
-        } else {
+        }
+        else {
             $type = 'down';
             $reason = $response['status_reason'];
 
+            // use current time as a starting point when an outage starts
             $data = ['device_id' => $device['device_id'],
-                'going_down' => strtotime($device['last_polled']), ];
-            dbInsert($data, 'device_outages');
+                'going_down' => time(), ];
+            if ( (! $maintenance && $mode) || ($maintenance && ! $mode) || (! $maintenance && ! $mode)) {
```

Comment:
seems more correct to me, but might be wrong:

`if ( ! $mode || (!$maintenance && $mode)) {`

`if ( ($maintenance xor $mode) || (! $maintenance && ! $mode)) {`

Reply:
That should work since that also covers all three cases